securedrop_client/gui/actions.py

```diff
@@ -126,3 +133,60 @@ def _on_confirmation_dialog_accepted(self) -> None:
                 return
             self.controller.delete_conversation(self.source)
             self._state.remove_conversation_files(id)
+
+
+class PrintConversationAction(QAction):  # pragma: nocover
+    def __init__(
+        self,
+        parent: QMenu,
+        controller: Controller,
+        source: Source,
+        export_service: Optional[export.Service] = None,
+    ) -> None:
+        """
+        Given some text and a reference to the controller, make something to display a file.
+        """
+        text = _("Print Conversation Transcript")
+
+        super().__init__(text, parent)
+
+        self.controller = controller
+        self._source = source
+
+        if export_service is None:
+            # Note that injecting an export service that runs in a separate
+            # thread is greatly encouraged! But it is optional because strictly
+            # speaking it is not a dependency of this FileWidget.
+            export_service = export.Service()
+
+        self._export_device = ConversationExportDevice(controller, export_service)
+
+        self.triggered.connect(self._on_triggered)
+
+    def _on_triggered(self) -> None:
+        """
+        Called when the print conversation action is clicked.
+        """
```

Comment:
Yeah, that follows what was in place / around. I do agree, it seems more useful to document what happens (either thay or remove the docstring if there isn't much to give insight about). Will look into that.

Reply:
Given the somewhat heavy naming convention, I don't think that saying _when_ the slot is called is very useful (in this case the name suggests strongly that is is when the action _triggered_ signal is emitted). I documented _what_ it does, though.

And declared it a Qt slot, because that's what it is. (It is connected to a Qt signal.)